### ISSUE TITLE
Add support of network music stream (such as online radio) with dynamic info about track

### ISSUE THREAD

**Issue by oleg-babintsev** [jacquesh/foo_openlyrics#49]
Add support of network music stream (such as online radio) with dynamic info about track.

![image](https://user-images.githubusercontent.com/2814327/129707060-532e33dc-c6c1-4a26-938a-7f0aad7de6ac.png)

**Comment by jacquesh** [comment]
None of the music stream/online radio sources I tried seemed to provide tag info in foobar. Can you confirm that tag info is available for you without any other components/plugins that might retrieve it?

Also could you give me an example of exactly which streams/radios you're using that provide this info?

Finally, in case I can't reproduce it, can you post a screenshot of the tag info that foobar has for that playlist item? (Right-click on it -> Properties -> Metadata tab).

**Comment by Big-aap** [comment]
https://s4-webradio.antenne.de/antenne/stream

This stream shows song title & artist in foobar. The metadata fields are mostly empty tho.
https://i.imgur.com/n3SFHWW.png

**Comment by oleg-babintsev** [comment]
1. I use RadioTunes service
2. Metadata don't contain dynamic information 
![image](https://user-images.githubusercontent.com/2814327/129732696-f2e4a2c9-2a0f-4d5f-bb5f-b08351da4310.png)

But as example, old plugin Lyrics Show Panel v3 can works with streams:
![image](https://user-images.githubusercontent.com/2814327/129733159-aec86de9-a45e-476e-93e7-122e8c424a63.png)

**Comment by jacquesh** [comment]
Hmmm, odd. I still don't get any metadata in the track list or in LSP3. I'm intrigued and I have a reasonable guess as to what I'd need to do to get this to work but its obviously difficult to try out if I can't get a radio stream working on my machine >_<

Bear with me here, these are the steps I'm taking:
1. File->Add Location -> https://s4-webradio.antenne.de/antenne/stream
2. Play the new item in the playlist
3. It plays but I still just see "stream" as the title and nothing under artist/album like so:
![image](https://user-images.githubusercontent.com/4639662/129763751-c5b39330-41d5-441d-aed6-2fd7650c569d.png)

I get this same behaviour for every radio/stream I tried (I tried some random ones from around the internet, though it seems RadioTunes requires a subscription to use external players so I couldn't try that). Am I missing a step or do doing something wrong here? >_<

**Comment by oleg-babintsev** [comment]
Hi,

I have fresh installation of foobar v1.6.7 (and on previous versions I always had the similar behaviour).
Test of https://s4-webradio.antenne.de/antenne/stream with Lyrics Show Panel v3:
![image](https://user-images.githubusercontent.com/2814327/129912129-77b15459-9a37-44b0-a7dc-d2c7aa430df8.png)

All works fine :)

**Comment by anazzani** [comment]
Could it be related to _dynamic track titles_ setting? See [here](https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Preferences:Networking).

**Comment by oleg-babintsev** [comment]
Thanks anazzani,

I confirm, if option "Enable dynamic track titles" is off the artist and title didn't detected by foobar playlist and by LSP3.

![image](https://user-images.githubusercontent.com/2814327/130580200-1f8c50e3-3547-4cd3-a4b1-18c586f7a312.png)

**Comment by jacquesh** [comment]
fyi I've added this to the list for v1.4 (not the next release, the one after that), or at least to revisit it using the new metadb v2 available in upcoming fb2k v2.0 beta (https://wiki.hydrogenaud.io/index.php?title=Foobar2000:Development:SDK_2022-08-10_Release_Notes)

**Comment by jacquesh** [comment]
Just to update those watching & waiting for this feature: I know I said this would be done for v1.4 but I've moved it to v1.5 because 1.4 is a bugfix-only release that I wanted to get out soon (because it fixes a bug that could cause data loss for some lyrics) and supporting internet radio streams requires some significant changes to the way openlyrics manages things internally. On top of that I'm unlikely to have any time to work on openlyrics for the next little while so I didn't want to release a new version with higher-than-usual potential for bugs and then not be around to fix it if anything critical came up.

On the upside, I have made a start on supporting this and indeed the new APIs provided by the recent fb2k updates do make this much easier to do. So this will get done, you'll just need to wait for a slightly higher version number. Thank you all for your patience!
